src/main/java/org/cascadebot/cascadebot/data/objects/Tag.java

```diff
@@ -0,0 +1,99 @@
+/*
+  * Copyright (c) 2019 CascadeBot. All rights reserved.
+  * Licensed under the MIT license.
+ */
+
+package org.cascadebot.cascadebot.data.objects;
+
+import org.apache.commons.lang.math.NumberUtils;
+import org.cascadebot.cascadebot.commandmeta.CommandContext;
+import org.cascadebot.cascadebot.utils.FormatUtils;
+
+import java.time.OffsetDateTime;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Tag {
+
+    private static final Map<String, BiFunction<CommandContext, String[], String>> placeholders = new HashMap<>();
```

Comment:
I feel like these should be stored separately.

Reply:
If you have any suggestions I'm all ears :)

Reply:
Maybe a TagsPlaceholders class as having it in tags will crowd it.

Reply:
Could do, it would make sense if we're gonna do that to do it in an enum perhaps

Reply:
Probably